Core/Foundation/Routing/Response.php

```diff
@@ -0,0 +1,409 @@
+        return $this;
+    }
+
+    /**
+     * Get data to format it.
+     *
+     * @param string $key The key to retrieve in the data. If null (by default), retrieves all data array.
+     * @return array|mixed The whole array, or just the requested sub element.
+     */
+    final public function getContentData($key = null)
+    {
+        if ($key !== null) {
+            if (!array_key_exists($key, $this->contentData)) {
+                return null;
+            }
+            return $this->contentData[$key];
+        }
+        return $this->contentData;
+    }
+    
+    /**
+     * Set suggested response format.
+     *
+     * @param string $format
+     */
+    final public function setResponseFormat($format)
+    {
+        $this->responseFormat = $format;
+    }
+    
+    /**
+     * Get suggested response format.
+     *
+     * @return string
+     */
+    final public function getResponseFormat()
+    {
+        return $this->responseFormat;
+    }
+    
+    /**
+     * Set template engine (callable finetuned & ready to be executed)
+     *
+     * @param object $callable
+     */
+    final public function setTemplateEngine($callable)
+    {
+        $this->templateEngine = $callable;
+    }
+    
+    /**
+     * Get template engine
+     *
+     * @param Container &$container The Container needed to call ViewFactory
+     * @return object
+     */
+    final public function getTemplateEngine(Container &$container)
+    {
+        if (!$this->templateEngine) {
+            $this->setTemplateEngine(new ViewFactory($container, $this->getEngineName()));
+        }
+
+        return $this->templateEngine;
+    }
+
+    /**
+     * Set engine name
+     *
+     * @param string $engineName
+     */
+    public function setEngineName($engineName)
+    {
+        $this->engineName = $engineName;
+    }
+
+    /**
+     * Get engine name
+     *
+     * @return string
+     */
+    public function getEngineName()
+    {
+        return $this->engineName;
+    }
+
+    /**
+     * Set template
+     *
+     * @param string $template
+     */
+    public function setTemplate($template)
+    {
+        $this->template = $template;
+    }
+
+    /**
+     * Get template
+     *
+     * @return string
+     */
+    public function getTemplate()
+    {
+        return $this->template;
+    }
+
+    /**
+     * Set legacyControllerName
+     *
+     * @param string $legacyControllerName
+     */
+    public function setLegacyControllerName($legacyControllerName)
+    {
+        $this->legacyControllerName = $legacyControllerName;
+    }
+
+    /**
+     * Get legacyControllerName
+     *
+     * @return string
+     */
+    public function getLegacyControllerName()
+    {
+        return $this->legacyControllerName ? $this->legacyControllerName : 'AdminDashboard';
+    }
+
+    /**
+     * Set headerToolbarBtn
+     *
+     * @param array[array] $headerToolbarBtn
+     */
+    public function setHeaderToolbarBtn(array $headerToolbarBtn)
```

Comment:
This is extremely business specific, should not be in a class in `Foundation`.
Moreover, this seems to be used only for the admin interface, so why would this be in all responses?
Lastly, I don't see why we need to have such specific fields inside a `Response` object: if `headerToolbarBtn` stuff is needed by the view, it should be passed through `setContent` via the rendered template or an array or I don't know what. Keep in mind `Response` is just the transport layer, it is used to speak with HTTP, not the application.